### ISSUE TITLE
Use of expires_at in project members fails due to debug function

### ISSUE THREAD

**Issue by HontoNoRoger** [gitlabform/gitlabform#654]
**Describe the bug**

When using `expires_at` for a project membership, gitlabform fails with an exception:

```
$ gitlabform -n ALL_DEFINED
🏗 GitLabForm version: 3.8.0 
= the latest stable 😊
>>> Getting ALL groups and projects DEFINED in the configuration...
Traceback (most recent call last):
  File "/usr/local/bin/gitlabform", line 8, in <module>
    sys.exit(run())
  File "/usr/local/lib/python3.10/dist-packages/gitlabform/run.py", line 5, in run
    GitLabForm().run()
  File "/usr/local/lib/python3.10/dist-packages/gitlabform/__init__.py", line 360, in run
    projects, groups = self._show_header(
  File "/usr/local/lib/python3.10/dist-packages/gitlabform/__init__.py", line 5[90](https://gitlab.sirius.supplyon.com/sirius/platform/cicd/gitlab/gitlab-configuration/-/jobs/528042#L90), in _show_header
    self.groups_and_projects_filters.filter(groups, projects)
  File "/usr/local/lib/python3.10/dist-packages/gitlabform/lists/filter.py", line 25, in filter
    self.omit_empty_configs.filter(groups, projects)
  File "/usr/local/lib/python3.10/dist-packages/gitlabform/lists/filter.py", line 77, in filter
    if self._project_has_empty_effective_config(project):
  File "/usr/local/lib/python3.10/dist-packages/gitlabform/lists/filter.py", line [99](https://gitlab.sirius.supplyon.com/sirius/platform/cicd/gitlab/gitlab-configuration/-/jobs/528042#L99), in _project_has_empty_effective_config
    config_for_project = self.configuration.get_effective_config_for_project(
  File "/usr/local/lib/python3.10/dist-packages/gitlabform/configuration/projects.py", line 53, in get_effective_config_for_project
    debug("Project config: %s", to_str(project_config))
  File "/usr/local/lib/python3.10/dist-packages/gitlabform/util.py", line 7, in to_str
    return json.dumps(a_dict, sort_keys=True)
  File "/usr/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/usr/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/usr/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type date is not JSON serializable
```

The problem lies within a debug output function, trying to JSON serialize a `date` object for output.

Call flow:
site-packages/gitlabform/lists/filter.py line 99: `config_for_project = self.configuration.get_effective_config_for_project(`
site-packages/gitlabform/configuration/projects.py line 53: `debug("Project config: %s", to_str(project_config))`
site-packages/gitlabform/util.py line 7: `return json.dumps(a_dict, sort_keys=True)`

Configuration example:

```yaml
...
  "my/awesome/project":
    members:
      enforce: true
      users:
        project_185_bot_b7ee8bf65aa55c389a9827a974bab4be:
          access_level: 40
          expires_at: 2024-12-12
...
```

**GitLabForm version**

🏗 GitLabForm version: 3.8.0 = the latest stable 😊

**GitLab version**

v16.5.2-ee

**Comment by HontoNoRoger** [comment]
The same resulting error existed before: [!207](https://github.com/gitlabform/gitlabform/issues/207)
So this seems to be a regression because of the `debug()` and `to_str()` use.

**Comment by amimas** [comment]
Thanks for the report @HontoNoRoger. Just to confirm, the issue happens only if gitlabform is executed with the `--debug` flag, right?

There was a major rewrite from v1 to v2. Sounds like the original fix somehow was dropped and wasn't caught because of missing tests. Anyways, PRs are welcome. Are you able to help contribute a fix?

**Comment by HontoNoRoger** [comment]
Hi @amimas, thank you for getting back to this. The call to gitlabform happened without the `--debug` flag. You can see this in the first like of the output at the top. The problem lies within the call to `to_str()` used as a parameter to the debug() function, so it is always evaluated.

I'll try to contribute a fix, though I'm not sure when that will happen due to the current Chrismas time.

**Comment by amimas** [comment]
This is now available in [v3.9.0 release](https://github.com/gitlabform/gitlabform/releases/tag/v3.9.0). Thanks for your contribution.
